src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -3,12 +3,19 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, bool IsNullable)
+    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, bool IsNullable)
         : InputType(Name, IsNullable)
     {
+        public InputModelType(string name, string? ns, string? accessibility, string? deprecated, string? description, InputModelTypeUsage usage, IReadOnlyList<InputModelProperty> properties, InputModelType? baseModel, IReadOnlyList<InputModelType> derivedModels, string? discriminatorValue, string? discriminatorPropertyName, bool isNullable)
+            : this(name, ns, accessibility, deprecated, description, usage, properties, derivedModels, discriminatorValue, discriminatorPropertyName, isNullable)
+        {
+            BaseModel = baseModel;
+        }
```

Comment:
We do not have to change the first line to add a private setter to BaseModel property, we could do something like this:
![image](https://github.com/Azure/autorest.csharp/assets/10554446/b067a999-3902-49c8-91ee-597e9406b172)
this might reduce the amount of changes in this PR